### PR TITLE
Allow custom logger to be passed instead of the default console

### LINKED ISSUE #113: Allow custom logger instead of console
As of now, output is just written to `console.log` or `console.error`. For consistent application logging, it would be necessary to supply a custom logger, like an instance of Bunyan for example.

### PR DESCRIPTION
**Pull request by orangecms**
This resolves #113 :)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -67,6 +67,11 @@ export interface OriginConfig extends OriginParams {
     http?: OriginHttpConfig
 }
 
+export interface Logger {
+    error(d: any): any;
+    log(d: any): any;
```
[the remark this answers is not in the document]
Comment (orangecms) [reply src/index.ts]: Yea I could not find the sources or documentation for the Go binary, so I am assuming to receive arbitrary input here. I didn't want to rely on guesses. Can you provide details on the data structures? And is the actual proxy going to be open source? That would really help, especially for auditing. Thank you! :)

**Comment by orangecms** [comment]
I'll be looking into it. This is the first time I touched a TypeScript project to be honest, so I'm happy about the hint towards the `void` type. :D I'll simply use Sinon for spying on `console.log` / `console.error` if that's okay. That's not really hard, I'm very familiar with test doubles. :)

Thank you a lot for the feedback! 😃

**Comment by glasser** [comment]
@orangecms Thanks for this feature! I am simplifying how apollo-engine-js communicates with engineproxy in #132, including making the default no communication, subprocess writes directly to our stdout/stderr without any copying inside the Node process.

Because of this, apollo-engine-js no longer really wanted to use console.log/console.error, so #132 will remove the `logger` option you added in this PR and replace it with a few other things: a `proxyStdoutStream` option if you don't want the proxy's stdout to be written directly to your stdout, and a new `restarting` EventEmitter event which replaces the only call to `console.error`. There's also a matching `proxyStderrStream` to allow flexibility with the proxy's stderr too.

Please let me know if these new APIs won't enable you to meet the goals for which you submitted this PR!

**Comment by orangecms** [comment]
Thanks @glasser - that sounds just fine so far. My main issue was that the logs interfered with the logging we have in the service that provides our GraphQL API. The `makeExecutableSchema` function from `graphql-tools` [accepts a `logger` option](https://www.apollographql.com/docs/graphql-tools/generate-schema.html#makeExecutableSchema) as well for error logging, so I thought this would be a simple, consistent solution. I'll be happy to contribute further ideas.

**Pull request merged**